src/crimson/os/seastore/onode.h

```diff
@@ -14,8 +14,14 @@
 namespace crimson::os::seastore {
 
 struct onode_layout_t {
+  // around 350 bytes for fixed fields in object_info_t,
+  // the left are for the variable-sized fields like oid
+  static constexpr int MAX_OI_LENGTH = 1024;
```

Comment:
I think there are two considerations about the size of `onode_layout_t`:
1.  The size limit of the key(`ghobject_t`) and value(`onode_layout_t`) to not break the onode-tree node-level split/merge operations;
2. The current implementation for `onode_layout_t` doesn't prefer large size, because inplace update would require to imbed the entire `sizeof(onode_layout_t)` into the seastore delta;

For 1. according to https://github.com/ceph/ceph/blob/master/src/crimson/os/seastore/onode_manager/staged-fltree/node_layout.h#L414-L447. If correct, we would need to set a proper size upper-bounds so that:
4 * max-key-size + 2 * max-value-size < 4KiB (node size),  for example both set to 640B.

For 2. I think currently we might need to keep the size of `onode_layout_t` as small as possible.